xbmc/pvr/PVRManager.cpp

```diff
@@ -1647,18 +1650,21 @@ std::string CPVRManager::GetPlayingTVGroupName()
   return IsStarted() && m_guiInfo ? m_guiInfo->GetPlayingTVGroup() : "";
 }
 
-void CPVRManager::UpdateLastWatched(CPVRChannel &channel)
+void CPVRManager::UpdateLastWatched(const CPVRChannelPtr &channel)
 {
+  if (!channel)
```

Comment:
in the methods (this one and 2 or 3 more above) where the ref changed to a shared pointer, you can assert() instead of trying to handle it gracefully. it would have segfaulted before too.